File: tests/fixtures/embedding/sample.py
"""
Order processing module for the e-commerce platform.
"""

from dataclasses import dataclass
from typing import List, Optional
from decimal import Decimal


@dataclass
class LineItem:
    product_id: str
    price: Decimal
    quantity: int


@dataclass
class Order:
    customer_id: str
    items: List[LineItem]
    total: Decimal
    status: str = "pending"


class OrderProcessor:
    """Processes and validates customer orders."""

    def __init__(self, tax_rate: Decimal = Decimal("0.08")):
        self.tax_rate = tax_rate

    def calculate_total(self, items: List[LineItem]) -> Decimal:
        """Calculate order total before tax."""
        return sum(item.price * item.quantity for item in items)

    def calculate_tax(self, subtotal: Decimal) -> Decimal:
        """Calculate tax for the given subtotal."""
        return subtotal * self.tax_rate

    def process_order(self, customer_id: str, items: List[LineItem]) -> Order:
        """Create and validate a new order."""
        if not items:
            raise ValueError("Order must have at least one item")

        subtotal = self.calculate_total(items)
        tax = self.calculate_tax(subtotal)
        total = subtotal + tax

        return Order(
            customer_id=customer_id,
            items=items,
            total=total,
        )


def find_order(orders: List[Order], order_id: str) -> Optional[Order]:
    """Find an order by ID from a list."""
    for order in orders:
        if hasattr(order, 'id') and order.id == order_id:
            return order
    return None


class DiscountCalculator:
    """Calculates discounts based on customer tier."""

    TIER_DISCOUNTS = {
        "bronze": Decimal("0.05"),
        "silver": Decimal("0.10"),
        "gold": Decimal("0.15"),
    }

    @classmethod
    def get_discount(cls, tier: str) -> Decimal:
        return cls.TIER_DISCOUNTS.get(tier.lower(), Decimal("0"))

File: tests/fixtures/embedding/sample.java
package com.example.orders;

import java.util.List;
import java.math.BigDecimal;

/**
 * Order service for managing customer orders.
 */
public class OrderService {

    private final OrderRepository repository;
    private final PaymentGateway paymentGateway;

    public OrderService(OrderRepository repository, PaymentGateway paymentGateway) {
        this.repository = repository;
        this.paymentGateway = paymentGateway;
    }

    public Order createOrder(String customerId, List<LineItem> items) {
        if (items == null || items.isEmpty()) {
            throw new IllegalArgumentException("Order must have at least one item");
        }
        BigDecimal total = calculateTotal(items);
        Order order = new Order(customerId, items, total);
        return repository.save(order);
    }

    public Order getOrder(String orderId) {
        return repository.findById(orderId)
            .orElseThrow(() -> new OrderNotFoundException(orderId));
    }

    public void cancelOrder(String orderId) {
        Order order = getOrder(orderId);
        if (order.getStatus() == OrderStatus.SHIPPED) {
            throw new IllegalStateException("Cannot cancel shipped order");
        }
        order.setStatus(OrderStatus.CANCELLED);
        repository.save(order);
    }

    private BigDecimal calculateTotal(List<LineItem> items) {
        return items.stream()
            .map(item -> item.getPrice().multiply(BigDecimal.valueOf(item.getQuantity())))
            .reduce(BigDecimal.ZERO, BigDecimal::add);
    }

    /**
     * Inner class for order validation.
     */
    public static class OrderValidator {
        public boolean validate(Order order) {
            return order != null
                && order.getCustomerId() != null
                && !order.getItems().isEmpty();
        }
    }
}

interface OrderRepository {
    Order save(Order order);
    java.util.Optional<Order> findById(String id);
}

class LineItem {
    private String productId;
    private BigDecimal price;
    private int quantity;

    public String getProductId() { return productId; }
    public BigDecimal getPrice() { return price; }
    public int getQuantity() { return quantity; }
}
